packages/babel-plugin-transform-class-properties/test/fixtures/assumption-noUninitializedPrivateFieldAccess/static-private/output.js

```diff
@@ -0,0 +1,13 @@
+class A {
+  static dynamicCheck() {
+    babelHelpers.assertClassBrand(this, A, _x);
+    _x = babelHelpers.assertClassBrand(this, A, 2);
+    [babelHelpers.toSetter(babelHelpers.assertClassBrand(this, A, _ => _x = _))._] = [];
+  }
+  static noCheck() {
+    _x;
+    _x = 2;
+    [_x] = [];
+  }
+}
+var _x = 2;
```

Comment:
Without the assumption, this would be
```js
class A {
  static dynamicCheck() {
    babelHelpers.assertClassBrand(this, A, _x)._;
    _x._ = babelHelpers.assertClassBrand(this, A, 2);
    [babelHelpers.assertClassBrand(this, A, _x)._] = [];
  }
  static noCheck() {
    _x._;
    _x._ = 2;
    [_x._] = [];
  }
}
var _x = {
  _: 2
};
```

The "assigned in destructuring" case gets worse, but it's a very rare operation to do compared to normal usage of private fields.